agent/src/beerocks/slave/son_slave_thread.cpp

```diff
@@ -2105,6 +2105,14 @@ bool slave_thread::handle_cmdu_ap_manager_message(Socket *sd,
         notification_out->iface_mac()  = hostap_params.iface_mac;
         notification_out->client_mac() = notification_in->params().mac;
         notification_out->bssid()      = notification_in->params().bssid;
+        if (!notification_in->params().association_frame) {
+            LOG(DEBUG) << "no association frame";
+            notification_out->set_association_frame("");
+        } else {
+            notification_out->set_association_frame(
+                notification_in->params().association_frame,
+                strnlen(notification_in->params().association_frame, ASSOCIATION_FRAME_SIZE) + 1);
+        }
```

Comment:
Meh I hate all this useless copying around of notification messages...

Reply:
Changed it.